secretstores/azure/keyvault/keyvault.go

```diff
@@ -47,13 +48,36 @@ func (k *keyvaultSecretStore) Init(metadata secretstores.Metadata) error {
 		Values: metadata.Properties,
 	}
 
-	authorizer, err := settings.GetAuthorizer()
-	if err == nil {
-		k.vaultClient.Authorizer = authorizer
+	certFilePath := metadata.Properties[componentSPNCertificateFile]
+	certBytes := []byte(metadata.Properties[componentSPNCertificate])
+	certPassword := metadata.Properties[componentSPNCertificatePassword]
+	clientID := metadata.Properties[componentSPNClientID]
+	tenantID := metadata.Properties[componentSPNTenantID]
+
+	var authorizer autorest.Authorizer
+	var err error
+	if certFilePath != "" && len(certBytes) > 0 && certPassword != "" && clientID != "" && tenantID != "" {
```

Comment:
I reviewed the code again. we do not need this code change.

I think this current component code should support both non-MSI mode and MSI mode. so this is a bug when refactoring the code. Let's review it next Monday. 

This condition is covered by this

https://github.com/dapr/components-contrib/blob/cf077cca65694b53088dc726603d625f28135faa/secretstores/azure/keyvault/authutils.go#L33-L46

and this:

https://github.com/dapr/components-contrib/blob/cf077cca65694b53088dc726603d625f28135faa/secretstores/azure/keyvault/authutils.go#L143-L149